### DISCUSSION TITLE
What is the reason for 0.0.0.0 in ALLOWED_HOSTS?

### DISCUSSION THREAD

**Discussion by NewJorg** [cookiecutter/cookiecutter-django#3460] [Q&A]
does anyone know what the reason is for the `0.0.0.0` in the `ALLOWED_HOSTS` in the `local.py`?
https://github.com/cookiecutter/cookiecutter-django/blob/9c74f2076c76fbba7bc5fe9d7722c5f2129f64ea/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/local.py#L13-L14

For me it feels a bit like it does not have any effect. Maybe it was meant to allow all, but then it should be `*` according the the linked documentation.

I hope somebody knows about the intention of the value or knows what it does that I do not see.

**Comment by luzfcb** [comment]
0.0.0.0 configure Django to accept requests that coming from outside your
operating system, like other computer in your network.

https://stackoverflow.com/a/61096774


Em sex, 10 de dez de 2021 13:29, NewJorg ***@***.***>
escreveu:

**Reply by NewJorg** [reply]
but it said to me "DISALLOWED HOST" with `0.0.0.0` in it. By replacing it with  `*` in the `ALLOWED_HOSTS` it work like intended.
See [next answer in stackoverflow](https://stackoverflow.com/a/68826930/11999147)

The `0.0.0.0:8000` after `runserver` is a bind address to a socket that should listen to all ip addresses. That I already knew and it was the reason why I could imagine that it should allow all hosts.

The `ALLOWED_HOSTS` filters based on the HTTP `Host` header (so after the connection to the socket was established successfully)

But when the intention was to allow all hosts, then I should be able to make it working like intended :-)
